ruby/ql/lib/codeql/ruby/ast/internal/Variable.qll

```diff
@@ -406,7 +406,11 @@ import Cached
 
 /** Holds if this scope inherits `name` from an outer scope `outer`. */
 private predicate inherits(Scope::Range scope, string name, Scope::Range outer) {
-  (scope instanceof Ruby::Block or scope instanceof Ruby::DoBlock) and
+  (
+    scope instanceof Ruby::Block or
+    scope instanceof Ruby::DoBlock or
+    scope instanceof Ruby::Lambda
+  ) and
```

Comment:
`class DoBlock extends Block`, so we should be able to write:

```suggestion
  (scope instanceof Ruby::Block or scope instanceof Ruby::Lambda) and
```

Reply:
Those are classes from `TreeSitter.qll`, and not `AST.qll`, so I think we still need both cases.